template-compose/.github/workflows/template_review_pull_request.yml

```
@@ -0,0 +1,67 @@
+name: Review pull request
+
+on:
+  pull_request:
+    types: [ opened, reopened, ready_for_review, synchronize ]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  review_pull_request:
+    if: github.event.pull_request.draft == false
+    name: Review pull request
+    runs-on: ubuntu-latest
+    environment: staging
+    timeout-minutes: 30
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
+
+      - name: Cache Gradle
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches/modules-*
+            ~/.gradle/caches/jars-*
+            ~/.gradle/caches/build-cache-*
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '2.7'
```

I saw the earlier comment by  @kaungkhantsoe about caching, should it be change here as well?
```suggestion
     - name: Set up Ruby
        uses: ruby/setup-ruby@v1
        with:
           ruby-version: '2.7'
           bundler-cache: true
```